Automattic/wp-calypso
# composite checkout: Support Ebanx TEF payments

#### Changes proposed in this Pull Request

* Support bank transfer payments via Ebanx in composite checkout

To do:
- [x] Hook up submit function
- [x] Make sure validation and error messages work
- [x] Clean up styles of fields (done in #44806)
- [x] Figure out why submission doesn't succeed
- [x] Fix logo and styling of collapsed payment method

#### Testing instructions

See D47729-code for forcing this payment method from the sandbox, and make sure you have `BILLING_SANDBOX` enabled to use the Ebanx sandbox environment. Test customer data is available at https://developer.ebanx.com/docs/resources/testCustomerData/#brazilian-test-data. (Note that CEP is the postal code, and CPF is the tax ID.) Also be aware that it is normal for bank transfer payments (event test payments) to take a while; whatever action you take on the redirect page, it may look like the provisioning page is stuck - just wait a few minutes.

* Enter checkout somehow (signup, new purchase, renewal) with some combination of products (plan only, plan+domain, extra domain only, etc.)
* Proceed through checkout to the payment method step. Check that the bank transfer method is available (D47729-code will force this from your sandbox)
* Select this method and jerk test the crap out of it. Particularly important flows include:
    * Try to submit with incomplete form data (missing tax ID, missing parts of the address, missing the name, missing the bank)
    * Try to submit with a contact country code that isn't BR. (There is no country code in the payment method itself, since it has to be BR, so we send the contact country code to Ebanx. If it is not BR, check that we show a useful error message.)
    * Submit with correct information, but at the redirect page, do not authorize the txn. Check that we show a useful error message and that checkout doesn't get stuck.
    * Submit with correct information and authorize the charge at the redirect page. Check that we generate a receipt in payments admin, and that the metadata on the receipt looks right (payment partner should be Ebanx, and the "card last 4 and expiry" will be all zeros because that doesn't exist for bank transfers.)

![image](https://user-images.githubusercontent.com/9310939/89924564-d236c100-dbc7-11ea-98e1-91e495ea1f13.png)
